File: Best_Time_to_Buy_and_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
            int profit = 0;
            int min_pr = INT_MAX;
            for(int i = 0 ;i<prices.size(); i++){
                    int cost = prices[i] - min_pr;
                    min_pr = min(min_pr,prices[i]);
                    profit = max(profit ,cost);
            }
            return profit;
        
    }
};

File: Best_Time_to_Buy_and_Sell_Stock.java
//leetcode - Best Time to Buy and Sell Stock.

public class Best_Time_to_Buy_and_Sell_Stock {
    public static void main(String[] args) {
        int[]prices = {7,1,5,3,6,4};
        System.out.println(maxProfit(prices));
    }
    public static int maxProfit(int[] prices) {
        int n = prices.length;
        if(n<=1){
            return 0;
        }
        int min = prices[0];
        int profit = 0;
        for(int i = 0;i<n;i++){
            int current = prices[i];
            if(min>prices[i]){
                min = prices[i];
            }
            else{
                int max = current - min;
                if(max > profit){
                    profit = max;
                }
            }

        
    }
    return profit;
}
}
